packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -108,23 +104,6 @@ const IconSwitcher = ({ icon, ...iconProps }: IconSwitcherProps) => {
   }
   // Backwards compatibility for apps using the "icon name".
   switch (icon) {
-    // Application icons
-    case 'TreeStructureIcon':
-      return <TreeStructureIcon {...iconProps} />;
-    case 'UserFilledIcon':
-      return <UserFilledIcon {...iconProps} />;
-    case 'SpeedometerIcon':
-      return <SpeedometerIcon {...iconProps} />;
-    case 'TagMultiIcon':
-      return <TagMultiIcon {...iconProps} />;
-    case 'CartIcon':
-      return <CartIcon {...iconProps} />;
-    case 'BoxIcon':
-      return <BoxIcon {...iconProps} />;
-    case 'GearIcon':
-      return <GearIcon {...iconProps} />;
-    case 'ListWithSearchIcon':
-      return <ListWithSearchIcon {...iconProps} />;
```

Comment:
Are these icons we don't use anymore?

Reply:
These are icons for our internal apps. Before the Org-level release we used to reference icons by their name (hence the logic here), however now they are embedded into the app configs / menus.

The logic here is just for backwards compatibility. I refactored it a bit to code split it so that it's only loaded when actually needed.

Reply:
Yes, the code splitting is a great idea.
I was wondering about the icons which have been removed from the static import list but do not exists in the new lazy imports:
* TreeStructureIcon
* UserFilledIcon
* SpeedometerIcon
* TagMultiIcon
* CartIcon
* BoxIcon
* GearIcon
* ListWithSearchIcon

Reply:
Ah yes sorry. These can be removed as the internal apps not embed them (I made a PR about that last week).